File: exceptions/src/test/java/com/cedarsoft/exceptions/MessagesTest.java
package com.cedarsoft.exceptions;

import com.cedarsoft.test.utils.LocaleRule;
import org.junit.*;

import java.util.Locale;

import static org.fest.assertions.Assertions.assertThat;

/**
 * @author Johannes Schneider (<a href="mailto:js@cedarsoft.com">js@cedarsoft.com</a>)
 */
public class MessagesTest {
  @Test
  public void testBundles() throws Exception {
    Messages messages = new Messages( ApplicationExceptionTest.BUNDLE );
    assertThat( messages ).isNotNull();

    assertThat( messages.getBundleName() ).isEqualTo( ApplicationExceptionTest.BUNDLE );

    assertThat( messages.getString( "ERROR_1", Locale.GERMAN ) ).isEqualTo( "The Value 1 de: <{0}>" );
    assertThat( messages.getString( "ERROR_1", Locale.ENGLISH ) ).isEqualTo( "The Value 1 en: <{0}>" );

    assertThat( messages.getString( "ERROR_1", Locale.FRENCH ) ).isEqualTo( "The Value 1 default: <{0}>" );
  }

  @Rule
  public LocaleRule localeRule = new LocaleRule( Locale.FRENCH );
}

File: commons/i18n/src/jvmTest/kotlin/com/cedarsoft/i18n/MessagesTest.kt
package com.cedarsoft.i18n

import assertk.*
import assertk.assertions.*
import com.cedarsoft.test.utils.WithLocale
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.util.Locale


@WithLocale("fr_FR")
class MessagesTest {

  private val bundle: String = "com.cedarsoft.i18n.test.testmessages"

  @Test
  fun testBundles() {
    val messages = Messages(bundle)
    assertThat(messages).isNotNull()
    assertThat(messages.bundleName).isEqualTo(bundle)
    assertThat(messages["ERROR_1", Locale.GERMAN]).isEqualTo("The Value 1 de: <{0}>")
    assertThat(messages["ERROR_1", Locale.ENGLISH]).isEqualTo("The Value 1 en: <{0}>")
    assertThat(messages["ERROR_1", Locale.FRENCH]).isEqualTo("The Value 1 default: <{0}>")
  }
}

